### PR TITLE
Adds some detail logging to test shoot creation/reconciliation loops

### PR DESCRIPTION
**Pull request by dguendisch**
**How to categorize this PR?**
/area testing
/kind enhancement
/priority normal

**What this PR does / why we need it**:
Adds some details to logging in the integration tests of shoot creation and reconciliation so that the logs contain the reason why a shoot was not considered as successfully created or reconciled.

**Which issue(s) this PR fixes**:
Misleading and incomplete logging.

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm
/status author-action
/invite @schrodit

File: test/system/complete_reconcile/gardener_reconcile_test.go [hunk 1]
```diff
@@ -73,9 +73,14 @@ var _ = Describe("Shoot reconciliation testing", func() {
 					f.Logger.Debugf("last acted gardener version %s does not match current gardener version %s", shoot.Status.Gardener.Version, *gardenerVersion)
 					continue
 				}
-				if framework.ShootCreationCompleted(&shoot) {
+				var completed bool
+				var msg string
+				if completed, msg = framework.ShootCreationCompleted(&shoot); completed {
```
Comment (timebertt) [inline test/system/complete_reconcile/gardener_reconcile_test.go]:
```suggestion
				if completed, msg := framework.ShootCreationCompleted(&shoot); completed {
```

File: test/framework/gardener_utils.go [hunk 2]
```diff
@@ -347,10 +349,12 @@ func (f *GardenerFramework) WaitForShootToBeReconciled(ctx context.Context, shoo
 			return retry.MinorError(err)
 		}
 		shoot = newShoot
-		if ShootCreationCompleted(shoot) {
+		var completed bool
+		var msg string
+		if completed, msg = ShootCreationCompleted(shoot); completed {
```
Comment (timebertt) [inline test/framework/gardener_utils.go]:
```suggestion

		completed, msg := ShootCreationCompleted(shoot)
		if completed {
```

File: test/framework/gardener_utils.go [hunk 3]
```diff
@@ -305,10 +305,12 @@ func (f *GardenerFramework) WaitForShootToBeCreated(ctx context.Context, shoot *
 			return retry.MinorError(err)
 		}
 		*shoot = *newShoot
-		if ShootCreationCompleted(shoot) {
+		var completed bool
+		var msg string
+		if completed, msg = ShootCreationCompleted(shoot); completed {
```
Comment (timebertt) [inline test/framework/gardener_utils.go]:
```suggestion

		completed, msg := ShootCreationCompleted(shoot)
		if completed {
```

**Review by timebertt — COMMENTED** [review]
Very nice, thanks for the PR!
I have only nitpicking comments

File: test/system/complete_reconcile/gardener_reconcile_test.go (hunk 1 above)
Comment (dguendisch) [reply test/system/complete_reconcile/gardener_reconcile_test.go]: done

File: test/framework/gardener_utils.go (hunk 2 above)
Comment (dguendisch) [reply test/framework/gardener_utils.go]: done

File: test/framework/gardener_utils.go (hunk 3 above)
Comment (dguendisch) [reply test/framework/gardener_utils.go]: done

**Comment by dguendisch** [comment]
Thanks for checking, addressed comments, ptal.

**Review by timebertt — APPROVED** [review]
Thanks for addressing my comments!
/lgtm

**Pull request merged**